pkg/strategy/xgap/strategy.go

```diff
@@ -271,56 +282,82 @@ func (s *Strategy) makeSpread(ctx context.Context, bestBid, bestAsk types.PriceV
 	return err
 }
 
-func getAdjustPositionOrder(symbol string, tradingMarket types.Market, currentPosition *types.Position, bestBid, bestAsk types.PriceVolume) (orderForms []types.SubmitOrder) {
-	basePosition, avgCost := currentPosition.GetBaseAndAverageCost()
-	if basePosition.Eq(fixedpoint.Zero) {
+// buildAdjustPositionOrders returns the order forms that can adjust the current position if possible.
+// When in a short position, it will try to place a buy order.
+// When in a long position, it will try to place a sell order.
+// It's based on the spread and the condition of the current position.
+func buildAdjustPositionOrders(
+	symbol string,
+	tradingMarket types.Market,
+	balances types.BalanceMap,
+	currentPosition *types.Position,
+	bestBid, bestAsk types.PriceVolume,
+) (orderForms []types.SubmitOrder) {
+	positionSnapshot := getPositionSnapshot(currentPosition)
+	if positionSnapshot.IsClosed() {
 		return
 	}
-	inShortPosition := basePosition.Sign() < 0
-	if inShortPosition && bestAsk.Price.Compare(avgCost) < 0 {
+
+	tryPlaceOrder := false
+	var pv types.PriceVolume
+	var side types.SideType
+	if positionSnapshot.IsShort() && bestAsk.Price.Compare(positionSnapshot.AverageCost) < 0 {
+		// in short position and the best ask price is less than the average cost
 		log.Infof(
 			"in short position, best ask price %s < average cost %s",
 			bestAsk.Price.String(),
-			avgCost.String(),
+			positionSnapshot.AverageCost.String(),
 		)
-		orderForms = append(orderForms, types.SubmitOrder{
-			Symbol:      symbol,
-			Side:        types.SideTypeBuy,
-			Type:        types.OrderTypeLimit,
-			Quantity:    fixedpoint.Max(bestAsk.Volume, tradingMarket.MinQuantity),
-			Price:       bestAsk.Price,
-			Market:      tradingMarket,
-			TimeInForce: types.TimeInForceIOC,
-		})
-	} else if !inShortPosition && bestBid.Price.Compare(avgCost) > 0 {
+		tryPlaceOrder = true
+		pv = bestAsk
+		side = types.SideTypeBuy
+	} else if positionSnapshot.IsLong() && bestBid.Price.Compare(positionSnapshot.AverageCost) > 0 {
+		// in long position and the best bid price is greater than the average cost
 		log.Infof(
 			"in long position, best bid price %s > average cost %s",
 			bestBid.Price.String(),
-			avgCost.String(),
+			positionSnapshot.AverageCost.String(),
 		)
-		orderForms = append(orderForms, types.SubmitOrder{
+		tryPlaceOrder = true
+		pv = bestBid
+		side = types.SideTypeSell
+	}
+	if tryPlaceOrder {
```

Comment:
you can check if the pv is assigned instead of introducing a new var

Reply:
ok.
I add a `.IsZero()` method instead.